File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque
def solution(cacheSize, cities):
    answer = 0
    cache = deque()
    for city in cities:
        city = city.lower()
        if city in cache:
            cache.remove(city)
            cache.append(city)
            answer += 1
        else:
            if len(cache) == cacheSize and cache:
                cache.popleft()
            if cacheSize > 0:
                cache.append(city)
            answer += 5
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.swift
func solution(_ cacheSize:Int, _ cities:[String]) -> Int {
    
    if cacheSize == 0 {
        return cities.count * 5
    }
    
    var cache: [String:Int] = [:]
    var time = 0
    for (idx, c) in cities.enumerated() {
        let city = c.lowercased()
        if let _ = cache[city] {
            cache[city] = idx
            time += 1
        } else {
            if cache.count >= cacheSize {
                let deleteKey = cache.min { $0.value < $1.value }!.key
                cache[deleteKey] = nil
            }
            cache[city] = idx
            time += 5
        }
    }
    return time
}